src/engine/strat_engine/mdv.rs

```diff
@@ -0,0 +1,124 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Manage the linear volume that stores metadata on pool levels 5-7.
+
+use std::fs::{create_dir, OpenOptions, read_dir, remove_file, rename};
+use std::io::prelude::*;
+use std::os::unix::io::AsRawFd;
+use std::path::PathBuf;
+
+use devicemapper::{LinearDev, DM};
+use serde_json;
+use nix::unistd::fsync;
+
+use engine::engine::{HasUuid, FilesystemUuid};
+use engine::EngineResult;
+use engine::PoolUuid;
+use super::filesystem::{create_fs, mount_fs, unmount_fs, StratFilesystem};
+use super::serde_structs::{Isomorphism, FilesystemSave};
+
+// TODO: Monitor fs size and extend linear and fs if needed
+// TODO: Document format of stuff on MDV in SWDD (currently ad-hoc)
+
+#[derive(Debug)]
+pub struct MetadataVol {
+    dev: LinearDev,
+    mount_pt: PathBuf,
+}
+
+impl MetadataVol {
+    pub fn new(pool_uuid: &PoolUuid, dev: LinearDev) -> EngineResult<MetadataVol> {
+        try!(create_fs(dev.devnode().expect("must have a devnode").as_path()));
+        MetadataVol::setup(pool_uuid, dev)
+    }
+
+    pub fn setup(pool_uuid: &PoolUuid, dev: LinearDev) -> EngineResult<MetadataVol> {
+        let new_path = PathBuf::from(format!("/dev/stratis/.mdv-{}", pool_uuid.simple()));
```

Comment:
I don't think so (at least not yet) because it's a MDV-local implementation detail.